### PR TITLE
btc-spv transaction parsing

### PR DESCRIPTION
**Pull request by pathoam**
#### Summary of Changes
added transaction, input, output parsing and initial test (input/output correctness checking to come)
Fixes #

### CODE REVIEW

File: programs/btc_spv_api/src/spv_processor.rs [hunk 1]
```diff
@@ -163,4 +163,27 @@ mod test {
...
...ffa55e28ebbcf94453ff11acb57b4b427c8e429f3a2e5441d63db2afdcb350fbc8010000008a473044022036431306f32b48e3b892d3dd9aa4deec0f5ea1dbabfc106cbe7b44cdbad83ad3022026a19c9093f946a28daa57df2ac9583dc13519d3c6b852cc3bcd2c479565e0b601410467c6736641546582785ca6f0d4e56c08fe67ee882c7907bf16471c85daf96ef48ce804a10a142175ce7d3be7d3bdcf77c884795441aaad7554a92a1ab47f000affffffff0280969800000000001976a91459d3fad18e55f2ffccb6829e67e77cb0e5b8e4d388ac00943577000000001976a9142cb4491ea630d4d5938f340baad7e31e4942626688ac0000000001000000069d74a193c94fa2d9aa9ed6a5d534defa99fe009cd2785d72c358b17bd36763c0010000008b483045022100d9912fadefc20eeb2c59fd10848367542b4065f9da044841361d6c28280fcfd702205078310c595e9efb65670f82b8b1fc07d498df1d56ba74c2186ea0ee8fbc8c11014104f1a2ed99e816a18a92beb211e312b7d07d0a64b8c55739f6b308240b8eae077d9ce5c616c6c32a50d26c04746a67cc6bce7f3466caf153146e883baeb403bf3affffffffb8cba6398365c2667249c5714cfaf0bf71c62b17874db28c582be377e5a408ed010000008b483045022100b386d0a462c3d84dde82ece6e011917370add71d3ac9bbf1a7f6800803d27f7b022060ce9deeb23b08f1658df36fa2333403e83116cb0803919cfd0776e2641af1f5014104f1a2ed99e816a18a92beb211e312b7d07d0a64b8c55739f6b308240b8eae077d9ce5c616c6c32a50d26c04746a67cc6bce";
```
Comment (garious) [inline programs/btc_spv_api/src/spv_processor.rs]: Any luck with `include_bytes!()`?

Comment (pathoam) [reply programs/btc_spv_api/src/spv_processor.rs]: yeah its in the version I pushed earlier

File: programs/btc_spv_api/src/spv_processor.rs [hunk 2]
```diff
@@ -163,4 +164,33 @@ mod test {
...
+        let i: usize = 5;
+
+        let mut txdata = &testdatabytes[80..];
+
+        let vilen = measure_variable_int(&txdata[0..9]).unwrap();
+        let txnum = decode_variable_int(&txdata[0..9]).unwrap();
+
+        txdata = &txdata[vilen..];
+        let tx = Transaction::new(txdata.to_vec());
+        println!("tx_num: {}", txnum);
```
Comment (garious) [inline programs/btc_spv_api/src/spv_processor.rs]: @mvines, this snuck by nits.sh. A bug?

Comment (mvines) [reply programs/btc_spv_api/src/spv_processor.rs]: println! free is currently opt-in: https://github.com/solana-labs/solana/blob/86213d38fe66aa5aaa359d2daca53b5ed42274e9/ci/nits.sh#L20

Maybe we should flip that around

Comment (garious) [reply programs/btc_spv_api/src/spv_processor.rs]: Yeah, opt-out seems better.  I'll add an issue.  Very low priority.

Comment (mvines) [reply programs/btc_spv_api/src/spv_processor.rs]: 👍 cc: #5961

**Pull request merged**